src/main/java/seedu/duke/ui/Ui.java

```diff
@@ -0,0 +1,143 @@
+package seedu.duke.ui;
+
+import java.util.Scanner;
+
+import seedu.duke.task.Task;
+import seedu.duke.task.TaskList;
+
+/**
+ * Ui Class responsible for handling all Ui displaying related matters.
+ */
+public class Ui {
+    /* Boolean representing whether there is a loadingError */
+    protected boolean loadingError;
+
+    /**
+     * Creates a Ui object.
+     */
+    public Ui() {
+        this.loadingError = false;
+    }
+
+    /**
+     * Prints out the welcome message.
+     */
+    public void showWelcome() {
+        String logo = " ____        _        \n"
+                + "|  _ \\ _   _| | _____ \n"
+                + "| | | | | | | |/ / _ \\\n"
+                + "| |_| | |_| |   <  __/\n"
+                + "|____/ \\__,_|_|\\_\\___|\n";
+        System.out.println("Hello from\n" + logo);
+
+        System.out.println("Hello! I'm seedu.duke.Duke\n"
+                + "What can I do for you?\n");
+    }
+
+    /**
+     * Shows a loading error screen if there is an error.
+     */
+    public void showLoadingError() {
+        System.out.println("----- Loading Error!! -----\n");
+    }
+
+    /**
+     * Prints out the error message if there is an error.
+     *
+     * @param errorMessage The error message.
+     */
+    public void showError(String errorMessage) {
+        System.out.println(Style.INDENTATION + errorMessage + "\n");
+    }
+
+    /**
+     * Returns the command after reading it through Scanner.
+     *
+     * @return The command in a string representation.
+     */
+    public String readCommand() {
+        Scanner myObj = new Scanner(System.in);
+        String command = myObj.nextLine();
+
+        return command;
+    }
+
+    /**
+     * Prints out a line as part of the UI.
+     */
+    public void showLine() {
+        System.out.println("__________________________________________"
+                + "__________________________________________" + "\n");
+    }
+
+    /**
+     * Prints out good bye message.
+     */
+    public void showGoodbye() {
+        System.out.println(Style.INDENTATION + "Bye. Hope to see you again soon!\n");
+    }
+
+    /**
+     * Displays all the tasks in the list.
+     * @param tasks
+     */
+    public void showList(TaskList tasks) {
+        System.out.println(Style.INDENTATION + "Here are the tasks in your list:");
+        tasks.listTasks();
+        System.out.println("");
+    }
+
+    /**
+     * Displays task after it has been marked.
+     * @param task The task to be marked.
+     */
+    public void showMarkStatus(Task task) {
+        System.out.println(Style.INDENTATION + "Nice! I've marked this seedu.duke.task as done:");
+        System.out.println(Style.INDENTATION + Style.HALF_INDENTATION + task + "\n");
+    }
+
```

Comment:
Did you intend to add another empty line here? Perhaps one would look cleaner instead.